From 72956444e1a09aa7fb583c320e00b91c7c778260 Mon Sep 17 00:00:00 2001
From: Ian Dannapel <ian.dannapel@iris-sensing.com>
Date: Mon, 28 Mar 2022 10:26:17 +0200
Subject: [PATCH 21/33] imx8mp-irma6r2.dts: Set i2c to max frequency and enable
 bus 1 & 4

-devices on i2c4 support only SCL @100KHz
-other devices support SCL @384KHz

Max setting is 384KHz because of ERR007805: I2C: When the I2C clock speed is configured for 400 kHz, the SCL
low period violates the I2C spec of 1.3 uS min
Workaround: To meet the clock low period requirement in fast speed mode, SCL
must be configured to 384KHz or less.

Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
---
 .../boot/dts/freescale/imx8mp-irma6r2.dts     | 25 +++++++++++++++++--
 1 file changed, 23 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts b/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
index 6efd0e80c894..e111b15b9f5c 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-irma6r2.dts
@@ -149,8 +149,15 @@
 };
 
 
+&i2c1 {
+	clock-frequency = <384000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	status = "okay";
+};
+
 &i2c2 {
-	clock-frequency = <100000>;
+	clock-frequency = <384000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c2>;
 	status = "okay";
@@ -233,7 +240,7 @@
 };
 
 &i2c3 {
-	clock-frequency = <100000>;
+	clock-frequency = <384000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
@@ -346,6 +353,13 @@
 	};
 };
 
+&i2c4 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c4>;
+	status = "okay";
+};
+
 &uart1 { /* BT */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
@@ -544,6 +558,13 @@
 			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA			0x400001c3
 		>;
 	};
+	
+	pinctrl_i2c4: i2c4grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_I2C4_SCL__I2C4_SCL			0x400001c3
+			MX8MP_IOMUXC_I2C4_SDA__I2C4_SDA			0x400001c3
+		>;
+	};
 
 	pinctrl_pmic: pmicirq {
 		fsl,pins = <
-- 
2.25.1

